### PR TITLE
Ajout de la mise à jour par CircleCI / Data.gouv.fr

### PR DESCRIPTION
**Pull request by Flightan**

### CODE REVIEW

File: import/index.js [hunk 1]
```diff
@@ -1,30 +1,54 @@
...
-  file: name+ '.csv',
+  file: `${name}.csv`,
 }
 
-
-
 analyse(options, (err, stdout, stderr) => {
-  if(err) return console.error(err)
-  console.log('analyze Done')
+  if(err) {
```
Comment (MattiSG) [inline import/index.js]: Inconsistent whitespace.

File: import/index.js [hunk 2]
```diff
@@ -1,30 +1,54 @@
...
+    console.error(err)
+    process.exit(1)
+  }
 
-  console.log('create model')
+  console.log('analyze Done')
```
Comment (MattiSG) [inline import/index.js]: Inconsistent casing.

File: import/index.js [hunk 3]
```diff
@@ -1,30 +1,54 @@
...
+
+  files.push({name: 'meta', path: `${__dirname}/../data/${name}.csv`})
 
   model(files, (data) => {
-    fs.writeFileSync(__dirname + '/../data/sites.json', JSON.stringify({data, meta: {lastUpdated: new Date().toLocaleString()}}, null, 2), 'utf8')
+    const filePath = `${__dirname}/../data/${name}.json`
+    fs.writeFileSync(filePath, JSON.stringify({data, meta: {lastUpdated: new Date().toLocaleString()}}, null, 2), 'utf8')
+
+    if (!process.env.DATA_GOUV_API_KEY) {
+      console.log('data.gouv api key not found, finished')
```
Comment (MattiSG) [inline import/index.js]: > Be more descriptive about what has been done rather than “finished”?

File: import/index.js [hunk 4]
```diff
@@ -1,30 +1,54 @@
...
+      process.exit(0)
+    }
+
+    request
+      .post('http://www.data.gouv.fr/api/1/datasets/5805f1e2c751df2bb879df72/resources/234299a6-e9a4-4cf4-a220-681a94ac3e92/upload/')
```
Comment (MattiSG) [inline import/index.js]: Add a comment to explain where this URL comes from and how it can be maintained.

File: src/App/App-test.js [hunk 5]
```diff
@@ -16,7 +16,7 @@ const arraySite = require('../../test/resources/array-sites')
 describe("Apps", () => {
 
   before(() => {
-    return fetchMock.mock('https://raw.githubusercontent.com/sgmap/sites/master/data/sites.json', 'GET', httpSites)
+    return fetchMock.mock('http://www.data.gouv.fr/s/resources/caracteristiques-techniques-des-sites/20161018-145351/sites.json', 'GET', httpSites)
```
Comment (MattiSG) [inline src/App/App-test.js]: Use HTTPS.

File: src/App/App.js [hunk 6]
```diff
@@ -32,7 +32,7 @@ class App extends Component {
   }
 
   getAnalytics() {
-    return fetch(`https://raw.githubusercontent.com/sgmap/sites/master/data/sites.json`)
+    return fetch('http://www.data.gouv.fr/s/resources/caracteristiques-techniques-des-sites/20161018-145351/sites.json')
```
Comment (MattiSG) [inline src/App/App.js]: Use HTTPS.

File: circle.yml [hunk 7]
```diff
@@ -10,6 +12,10 @@ general:
 test:
   post:
     - cat ./coverage/lcov.info | ./node_modules/.bin/coveralls --verbose
+    - >
+      if [ -n "${RUN_UPDATE}" ]; then
```
Comment (MattiSG) [inline circle.yml]: What is the usage of that parameter? How am I supposed to use the `trigger_circleci_update.sh` script? Why not simply always update on merges to `master`?

File: circle.yml [hunk 8]
```diff
@@ -10,6 +12,10 @@ general:
...
+        npm run update;
```
Comment (MattiSG) [inline circle.yml]: > No need for a semicolon here.
> I recommend to either use full inline style (`if [ … ]; then …; fi`) or full multiline style:
> ```sh
> if [ … ]
> then …
> fi
> ```

File: circle.yml (hunk 7 above)
Comment (Flightan) [inline circle.yml]: ... 
why not indeed...

File: circle.yml [hunk 9]
```diff
@@ -12,10 +12,7 @@ general:
 test:
   post:
     - cat ./coverage/lcov.info | ./node_modules/.bin/coveralls --verbose
-    - >
-      if [ -n "${RUN_UPDATE}" ]; then
-        npm run update;
-      fi
+    - if [ $CIRCLE_BRANCH = 'master' ]; then npm run update; fi
```
Comment (MattiSG) [inline circle.yml]: This shouldn't be in the `test: post` section, that test is exactly what is done by Circle in the `deployment` section  :)

File: circle.yml [hunk 10]
```diff
@@ -10,6 +12,7 @@ general:
 test:
   post:
     - cat ./coverage/lcov.info | ./node_modules/.bin/coveralls --verbose
+    - if [ $CIRCLE_BRANCH = 'master' ]; then npm run update; fi
```
Comment (MattiSG) [inline circle.yml]: This shouldn't be in the `test: post` section, that test is exactly what is done by Circle in the `deployment` section  :)

**Comment by MattiSG** [comment]
There's an issue when deploying: https://circleci.com/gh/sgmap/verif.site/280.

**Comment by MattiSG** [comment]
Did the push work?

**Comment by Flightan** [comment]
Yes

**Pull request merged**